samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,114 @@ def _validate_deployment_preference_and_add_update_policy(
+
+        :param dict kwargs: already-converted resources that may need to be modified when converting this \
+        macro to pure CloudFormation
+        :returns: a list of vanilla CloudFormation Resources, to which this Function expands
+        :rtype: list
+        """
+        resources = []
+        syntheticsCanary = self._construct_synthetics_canary()
+        resources.append(syntheticsCanary)
+
+        return resources
+
+    def _construct_synthetics_canary(self):
+        """Constructs a AWS::Synthetics::Canary resource."""
+        canary = SyntheticsCanary(
+            self.logical_id, depends_on=self.depends_on, attributes=self.get_passthrough_resource_attributes()
+        )
+        canary.ArtifactS3Location = self.ArtifactS3Location
+        canary.Code = self._construct_code_dict()
+        canary.ExecutionRoleArn = self.Role
+        canary.FailureRetentionPeriod = self.FailureRetentionPeriod
+        # Make a default for canary name
+        canary.Name = self.FunctionName
+        canary.RuntimeVersion = self.Runtime
+        canary.Schedule = self.Schedule
+        canary.StartCanaryAfterCreation = self.StartCanaryAfterCreation
+        canary.SuccessRetentionPeriod = self.SuccessRetentionPeriod
+        canary.Tags = self._construct_tag_list(self.Tags)
+        canary.VPCConfig = self.VpcConfig
+
+        if self.Tracing or self.Environment or self.MemorySize or self.Timeout:
+            canary.RunConfig = self._construct_run_config()
+        return canary
+
+    def _construct_run_config(self):
+        runconfig = {"ActiveTracing": self.Tracing, "MemoryInMB": self.MemorySize, "TimeoutInSeconds": self.Timeout}
+        if self.Environment:
+            runconfig["EnvironmentVariables"] = self.Environment["Variables"]
+
+        return {k: v for k, v in runconfig.items() if v is not None}
+
+    def _construct_code_dict(self):
+        """Constructs Synthetics Canary Code Dictionary based on the accepted SAM artifact properties such
+        as `InlineCode` and `CodeUri` and also raises errors if more than one of them is
+        defined. `PackageType` determines which artifacts are considered.
+
+        :raises InvalidResourceException when conditions on the SAM artifact properties are not met.
+        """
+        # accepted artifacts
+        artifacts = {"InlineCode": self.InlineCode, "CodeUri": self.CodeUri}
+
+        # Inline function for transformation of inline code.
+        # It accepts arbitrary argumemnts, because the arguments do not matter for the result.
+        def _construct_inline_code(*args, **kwargs):
+            return {"Handler": self.Handler, "Script": self.InlineCode}
+
+        # dispatch mechanism per artifact on how it needs to be transformed.
+        artifact_dispatch = {
+            "InlineCode": _construct_inline_code,
+            "CodeUri": construct_s3_location_object,
+        }
+
+        filtered_artifacts = dict(filter(lambda x: x[1] != None, artifacts.items()))
```

Comment:
Sorry I am not able to clearly understand what is going on here? Is it similar to logic above where you are filtering out fields from dict with value None?

Reply:
It does the same thing, instead of using `dict(filter(lambda x: x[1] != None, artifacts.items()))`